nautobot/core/management/commands/send_installation_metrics.py

```diff
@@ -0,0 +1,70 @@
+import hashlib
+import platform
+import requests
+import uuid
+
+from constance import config
+from django.apps import apps
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from nautobot.utilities.config import get_settings_or_config
+
+
+class Command(BaseCommand):
+    help = "Send installation metrics for this Nautobot installation."
+
+    def _hash(self, plaintext):
+        return hashlib.sha256(plaintext.encode("utf8")).hexdigest()
+
+    def get_hashed_plugins_with_version(self):
+        plugins = {}
+        for plugin_name in settings.PLUGINS:
+            plugin_name = plugin_name.rsplit(".", 1)[-1]
+            plugin_config = apps.get_app_config(plugin_name)
+            plugins[self._hash(plugin_name)] = getattr(plugin_config, "version", None)
+        plugins = dict(sorted(plugins.items()))
+
+        return plugins
+
+    def handle(self, *args, **options):
+        # skip if metrics are disabled
+        if settings.INSTALLATION_METRICS_ENABLED is not True:
+            return
+
+        # get the deployment id for this install from constance or settings
+        # if one is not already set, generate a random uuid and set it in constance
+        deployment_id = get_settings_or_config("DEPLOYMENT_ID")
+        if not deployment_id:
+            deployment_id = str(uuid.uuid4())
+            config.DEPLOYMENT_ID = deployment_id
+
+        # build the json payload to send
+        payload = {
+            "deployment_id": deployment_id,
+            "nautobot_version": settings.VERSION,
+            "python_version": platform.python_version(),
+            "installed_apps": self.get_hashed_plugins_with_version(),
+            "debug": settings.DEBUG,
+        }
+
+        # send the payload to the metrics endpoint
+        metrics_endpoint = "https://nautobot.cloud/api/nautobot/installation-metric/"
+
+        try:
+            prepared_request = requests.Request("POST", metrics_endpoint, json=payload).prepare()
+        except requests.exceptions.RequestException as e:
+            print(f"Error forming HTTP request: {e}")
+            return
```

Comment:
Is this something you've encountered?

Should we fail the command if this happens, or is it acceptable/preferable to have it still considered as successful in case of temporary outages on our end?

Reply:
As written we shouldn't encounter a request exception. If the url or json payload is changed to invalid data this would catch that. It's probably ok to remove?

```py
>>> requests.Request("POST", "localhost", json={}).prepare()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-DIHtVAwE-py3.10/lib/python3.10/site-packages/requests/models.py", line 299, in prepare
    p.prepare(
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-DIHtVAwE-py3.10/lib/python3.10/site-packages/requests/models.py", line 368, in prepare
    self.prepare_url(url, params)
  File "/home/gary/.cache/pypoetry/virtualenvs/nautobot-DIHtVAwE-py3.10/lib/python3.10/site-packages/requests/models.py", line 439, in prepare_url
    raise MissingSchema(
requests.exceptions.MissingSchema: Invalid URL 'localhost': No scheme supplied. Perhaps you meant https://localhost?
```